### PR TITLE
Move project selection to deployment configuration

### LINKED ISSUE #675: Move cloud configuration project selector to deployment

### PR DESCRIPTION
**Pull request by etanshaul**
Fixes #675 

@chanseokoh please take a look.

This moves the project selection from the cloud configuration to the deployment configuration. This way individual deployments can target different cloud projects without having to create a new cloud configuration.

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineServerConfiguration.java [hunk 1]
```diff
@@ -32,32 +28,17 @@
     ServerConfigurationBase<AppEngineServerConfiguration> {
 
   private String cloudSdkHomePath;
-  private String cloudProjectName;
   private String googleUserName;
```
Comment (chanseokoh) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineServerConfiguration.java]: Seems like `googleUserName` along with the getter/setter is never used.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java [hunk 2]
```diff
@@ -265,14 +253,19 @@ private void setDeploymentSourceName(String filePath) {
...
           GctBundle.message("appengine.flex.config.user.specified.artifact.error"));
     } else if (!isUserSpecifiedPathDeploymentSource() && !deploymentSource.isValid()) {
       throw new ConfigurationException(
           GctBundle.message("appengine.config.deployment.source.error"));
-    } else if(versionOverrideCheckBox.isSelected() && StringUtils.isBlank(versionIdField.getText())) {
+    } else if (StringUtils.isBlank(projectSelector.getText())) {
+      throw new ConfigurationException(
+          GctBundle.message("appengine.flex.config.project.missing.message"));
+    } else if (versionOverrideCheckBox.isSelected() && StringUtils
+        .isBlank(versionIdField.getText())) {
```
Comment (chanseokoh) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java]: Line-breaking seems unusual.

**Comment by etanshaul** [comment]
@chanseokoh updated.

**Comment by chanseokoh** [comment]
LGTM

**Pull request merged**